pkg/ruler/api_test.go

```diff
@@ -27,125 +27,219 @@ import (
+			configuredRules: rulespb.RuleGroupList{
+				&rulespb.RuleGroupDesc{
+					Name:      "group1",
+					Namespace: "namespace1",
+					User:      "user1",
+					Rules:     []*rulespb.RuleDesc{mockRecordingRuleDesc("UP_RULE", "up"), mockAlertingRuleDesc("UP_ALERT", "up < 1")},
+					Interval:  interval,
+				},
+			},
+			overrides: map[string]rulerLimits{
 				"user1": {
-					&rulespb.RuleGroupDesc{
-						Name:          "group1",
-						Namespace:     "namespace1",
-						User:          "user1",
-						SourceTenants: []string{"tenant-1"},
-						Rules: []*rulespb.RuleDesc{
-							{
-								Record: "UP_RULE",
-								Expr:   "up",
-							},
-							{
-								Alert: "UP_ALERT",
-								Expr:  "up < 1",
-							},
+					recordingRulesEvaluationEnabled: false,
+					alertingRulesEvaluationEnabled:  true,
+				},
+			},
+			expectedRules: []*RuleGroup{
+				{
+					Name: "group1",
+					File: "namespace1",
+					Rules: []rule{
+						&alertingRule{
+							Name:   "UP_ALERT",
+							Query:  "up < 1",
+							State:  "inactive",
+							Health: "unknown",
+							Type:   "alerting",
+							Alerts: []*Alert{},
 						},
-						Interval: interval,
 					},
+					Interval: 60,
 				},
 			},
-			expectedResponse: response{
-				Status: "success",
-				Data: &RuleDiscovery{
-					RuleGroups: []*RuleGroup{
+		},
+		"should load and evaluate no rules if rules evaluation is disabled for the tenant": {
+			configuredRules: rulespb.RuleGroupList{
+				&rulespb.RuleGroupDesc{
+					Name:      "group1",
+					Namespace: "namespace1",
+					User:      "user1",
+					Rules:     []*rulespb.RuleDesc{mockRecordingRuleDesc("UP_RULE", "up"), mockAlertingRuleDesc("UP_ALERT", "up < 1")},
+					Interval:  interval,
+				},
+			},
+			overrides: map[string]rulerLimits{
+				"user1": {
+					recordingRulesEvaluationEnabled: false,
+					alertingRulesEvaluationEnabled:  false,
+				},
+			},
+			expectedRules: []*RuleGroup{},
+		},
+		"should load and evaluate the configured rules with special characters": {
+			configuredRules: rulespb.RuleGroupList{
+				&rulespb.RuleGroupDesc{
+					Name:      ")(_+?/|group1+/?",
+					Namespace: ")(_+?/|namespace1+/?",
+					User:      "user1",
+					Rules: []*rulespb.RuleDesc{
+						{
+							Record: "UP_RULE",
+							Expr:   "up",
+						},
 						{
-							Name:          "group1",
-							File:          "namespace1",
-							SourceTenants: []string{"tenant-1"},
-							Rules: []rule{
-								&recordingRule{
-									Name:   "UP_RULE",
-									Query:  "up",
-									Health: "unknown",
-									Type:   "recording",
-								},
-								&alertingRule{
-									Name:   "UP_ALERT",
-									Query:  "up < 1",
-									State:  "inactive",
-									Health: "unknown",
-									Type:   "alerting",
-									Alerts: []*Alert{},
-								},
-							},
-							Interval: 60,
+							Alert: "UP_ALERT",
+							Expr:  "up < 1",
+						},
+					},
```

Comment:
```suggestion
					Rules:     []*rulespb.RuleDesc{mockRecordingRuleDesc("UP_RULE", "up"), mockAlertingRuleDesc("UP_ALERT", "up < 1")},
					Interval:  interval,
```

not sure if you consider this out of scope